src/core/resolver/utils.ts

```diff
@@ -81,9 +85,11 @@ export const resolveContentNode = (
 
       // Data source might be preset on a referenced attribute in the uidl node
       // ex: attrs[options] in case of a dropdown primitive with select/options
-      if (typeof dataSource === 'string' && dataSource.startsWith('$attrs.') && node.attrs) {
-        const nodeDataSourceAttr = dataSource.replace('$attrs.', '')
+      if (dataSource.type === 'dynamic' && dataSource.content.referenceType === 'attr') {
```

Comment:
is this for the special case of mapping different attribute names to other attributes? The one that is used in mappings?

Reply:
yes. referenceType = 'attr' will only be handled by the resolver, as it references another attribute in the UIDL node. later on, it might even reference another attribute from another element, but for now it's good enough